20240129-satellite-managed-encryption.md

```diff
@@ -0,0 +1,124 @@
+tags: ["satellite", "satellite/console", "encryption"]
+---
+
+# Satellite-Managed Encryption Passphrases
+
+## Essentials
+
+### Header
+
+Date: 2024-01-29
+
+Owner: mobyvb
+
+Accountable:
+- Console squad
+
+Consulted:
+- Product
+- Delivery (TODO)
+- Satellite (TODO)
+
+Informed:
+- Engineering
+
+### Context
+
+One common issue for new users is the added step of managing an encryption passphrase for a project. The user must store the encryption passphrase, and enter it when generating access credentials or using the object browser in the Satellite UI.
+
+Having a user-managed encryption passphrase has privacy benefits for the user, but it adds an additional layer of complexity to user experience. We want to continue to support users who want to manage their own encryption credentials, while allowing other users who are not interested in managing their own encryption credentials to easily use the product. 
+
+Additionally, by default, object paths are encrypted. This has privacy benefits, but prevents certain features that some users are interested in, like lexicographic ordering of object keys. Like with user-managed encryption passphrases, we want to continue supporting path encryption for users who are interested, while allowing other users who want lexicographic ordering features to create a project which has path encryption turned off.
+
+### Goals
+
+Users are provided the option to create a project that uses "satellite-managed encryption". One of these projects:
+* uses a securely-stored, random passphrase stored on the satellite as the project's encryption passphrase
+* restricts all files to be uploaded without path encryption
+
+The main goals are to:
+* provide users with an experience that does not require managing and entering an encryption passphrase in the Satellite UI
+* provide users with an easy way to view and list files in lexographical order
+
+### Approach / Design
+
+#### Key Management Store
+
+We need a secure way to store and look up project encryption passphrases on the Satellite. Someone with normal Satellite DB access should not be able to view this data, and ideally it is encrypted. At the same time, it must be very reliable, as losing data about project encryption passphrases means that data may have been lost.
+
+The specific technology/product used for the KMS can be decided in a later stage. The primary features and concerns we should keep in mind:
```

Comment:
@wilfred-asomanii has updated the document with info about the KMS

Reply:
Vault sounds good and we have mentioned a few times that it would be nice to have to store all our secrets.

However,  in my opinion, deploying and integrating it in the satellite should have a separate design doc because of what's commented in this design doc about it.

Reply:
Vault isn't the selected option anymore.